src/decode.rs

```diff
@@ -174,22 +182,19 @@ impl<'a, I: AsRef<[u8]>> DecodeBuilder<'a, I> {
     ///     output);
     /// ```
     pub fn into<O: AsMut<[u8]>>(self, mut output: O) -> Result<usize> {
-        if self.check {
+        match self.check {
+            Check::Disabled => decode_into(self.input.as_ref(), output.as_mut(), self.alpha),
             #[cfg(feature = "check")]
-            {
-                decode_check_into(
-                    self.input.as_ref(),
-                    output.as_mut(),
-                    self.alpha,
-                    self.expected_ver,
-                )
+            Check::Enabled => {
+                decode_check_into(self.input.as_ref(), output.as_mut(), self.alpha, None)
             }
-            #[cfg(not(feature = "check"))]
-            {
-                unreachable!("This function requires 'check' feature")
-            }
-        } else {
-            decode_into(self.input.as_ref(), output.as_mut(), self.alpha)
+            #[cfg(feature = "check")]
+            Check::EnabledVer(expected_ver) => decode_check_into(
```

Comment:
```suggestion
            Check::EnabledVersion(expected_ver) => decode_check_into(
```

Though it might be better to just have a variant `Enabled(Option<u8>)` instead of splitting this into two variants.

Reply:
Yes, `Enabled(Option<u8>)` looks better for sure, but then total enum size will be 3 bytes instead 2...

Reply:
It _should_ be optimized to 2 still, Rust can choose different discriminators for the enum to merge the one for `Option` and `Check` into a single byte. [Checking in the playground agrees](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a945819ef23ae7f927edc9e56847e721).

Reply:
Oh, it's really 2 bytes. I checked before with `size_of`, but receive 3, not sure why and code already removed.
Thanks, I'll fix shortly.

Reply:
Fixed